roles/splunk_common/tasks/set_as_hec_receiver.yml

```diff
@@ -1,46 +1,37 @@
 ---
 - name: Enable HEC services
-  uri:
-    url: "{{ cert_prefix }}://127.0.0.1:{{ splunk.svc_port }}/servicesNS/nobody/splunk_httpinput/data/inputs/http/http/enable"
-    method: POST
-    user: "{{ splunk.admin_user }}"
-    password: "{{ splunk.password }}"
-    validate_certs: false
-    status_code: 200
-    timeout: 10
+  ini_file:
+    path: "{{ splunk_hec_inputs_app_path }}/local/inputs.conf"
+    section: http
+    option: disabled
+    value: 0
+    mode: '0600'
+    backup: no
   ignore_errors: true
   when: not splunk.hec_disabled|bool
+  register: splunk_hec_input
   no_log: "{{ hide_password }}"
 
 - name: Create new HEC token
-  uri:
-    url: "{{ cert_prefix }}://127.0.0.1:{{ splunk.svc_port }}/servicesNS/nobody/splunk_httpinput/data/inputs/http"
-    method: POST
-    user: "{{ splunk.admin_user }}"
-    password: "{{ splunk.password }}"
-    validate_certs: false
-    body:
-      name: "splunk_hec_token"
-      token: "{{ splunk.hec_token }}"
-    body_format: "form-urlencoded"
-    status_code: 201,409
-    timeout: 10
+  ini_file:
```

Comment:
If you move to `ini_file`, we either need to do this step before Splunk starts, or you need to add a `notify` trigger here as to restart Splunk after the provisioning has completed. I'd prefer the former because unnecessary restarts add delay to the time-to-value from the moment users start their containers to when they can actually begin using Splunk.